lib/validates_email_format_of.rb

```diff
@@ -297,11 +297,15 @@ def self.validate_domain_part_syntax(domain, idn: true)
 
   def self.deprecation_warn(msg)
     if defined?(ActiveSupport::Deprecation)
-      ActiveSupport::Deprecation.warn(msg)
+      deprecator.warn(msg)
     else
       warn(msg)
     end
   end
+
+  def self.deprecator
```

Comment:
It's hilarious that they've deprecated their deprecator.

Is there any way we can support both the new and old styles together and keep old Rails support going?  I'm ok to drop old versions if it becomes really onerous, but I wonder if there's a simple fix that keeps everything working.

I think this is the commit where things changed...
https://github.com/rails/rails/commit/c682bf26417f2a88b0efa46499e94dcf45955f4f

Reply:
yes, we can work around the Rails API detail with conditions.

But, the harder problem was that setup-ruby failed to install gems on a more basic level, when the `bundler: 1` was configured.

Reply:
I don't have the time I wish I had to troubleshoot CI.  

I'm ok to drop the old CI matrix versions for now just to get this out.  Would you be able to add the compatibility conditional to make sure we still work on older Rails?  Then I'll get this merged and out the door.

Reply:
😅 It may take some time, but we'll get there.